### PR TITLE
feat: [CDS-54329]: Lambda Alias support changes

### PR DESCRIPTION
**Pull request by sainath-b**
## Describe your changes

Native Lambda NG Support - Alias create update changes in deploy and rollback

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)

### CODE REVIEW

**Review by rohitk-h — APPROVED** [review]
> Reviewed 13 of 13 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by rohitk-h — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 13 file(s) to review. 1 of 2 approvals obtained.

**Review by rohitk-h — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 13 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 1]
```diff
@@ -252,13 +260,16 @@ private TaskChainResponse executePrepareRollbackTask(Ambiance ambiance, StepElem
...
+        getManifestContentsOfManifestType(awsLambdaStepPassThroughData, ManifestType.AwsLambdaFunctionDefinition);
+
     AwsLambdaPrepareRollbackRequest awsLambdaPrepareRollbackRequest =
         AwsLambdaPrepareRollbackRequest.builder()
             .awsLambdaCommandTypeNG(AwsLambdaCommandTypeNG.AWS_LAMBDA_PREPARE_ROLLBACK)
             .commandName(AWS_LAMBDA_PREPARE_ROLLBACK_COMMAND_NAME)
             .commandUnitsProgress(UnitProgressDataMapper.toCommandUnitsProgress(unitProgressData))
             .awsLambdaInfraConfig(getInfraConfig(infrastructureOutcome, ambiance))
-            .awsLambdaDeployManifestContent(awsLambdaStepPassThroughData.getManifestContent())
+            .awsLambdaDeployManifestContent(awsLambdaFunctionDefinitionContents.get(0))
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:272]: Can we add a check here before referencing index 0

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 2]
```diff
@@ -267,9 +278,41 @@ private TaskChainResponse executePrepareRollbackTask(Ambiance ambiance, StepElem
...
-    String manifestContent = gitTaskResponse.getGitFetchFilesResults().get(0).getFiles().get(0).getFileContent();
-    return engineExpressionService.renderExpression(ambiance, manifestContent);
+  private List<String> getManifestContentsOfManifestType(
+      AwsLambdaStepPassThroughData awsLambdaStepPassThroughData, String manifestType) {
+    List<ManifestOutcome> manifestOutcomes = awsLambdaStepPassThroughData.getManifestsOutcomes();
+
+    List<String> manifestContentsOfManifestType = new ArrayList<>();
+
+    for (ManifestOutcome manifestOutcome : manifestOutcomes) {
+      if (manifestOutcome.getType() == manifestType) {
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:288]: use equals method

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 3]
```diff
@@ -267,9 +278,41 @@ private TaskChainResponse executePrepareRollbackTask(Ambiance ambiance, StepElem
...
+        manifestContentsOfManifestType.add(
+            awsLambdaStepPassThroughData.getManifestFileContentsMap().get(manifestOutcome.getIdentifier()).get(0));
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:290]: again a check before referencing index 0

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 4]
```diff
@@ -457,16 +507,23 @@ public TaskChainResponse executeTask(Ambiance ambiance, StepElementParameters st
...
+        getManifestContentsOfManifestType(awsLambdaStepPassThroughData, ManifestType.AwsLambdaFunctionAliasDefinition);
+
     AwsLambdaDeployRequest awsLambdaDeployRequest =
         AwsLambdaDeployRequest.builder()
             .awsLambdaCommandTypeNG(AwsLambdaCommandTypeNG.AWS_LAMBDA_DEPLOY)
             .commandName(AWS_LAMBDA_DEPLOY_COMMAND_NAME)
             .commandUnitsProgress(UnitProgressDataMapper.toCommandUnitsProgress(unitProgressData))
             .awsLambdaInfraConfig(getInfraConfig(infrastructureOutcome, ambiance))
-            .awsLambdaDeployManifestContent(awsLambdaStepPassThroughData.getManifestContent())
+            .awsLambdaDeployManifestContent(awsLambdaFunctionDefinitionContents.get(0))
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:522]: check for size of list here as well

**Review by sainath-b — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 13 file(s) to review. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 272 at r3](https://reviewable.io/reviews/harness/harness-core/44560#-NPSxeZubaCsQcvOcKf7-r3-272:-NPXV5V7CUKvCJVA6oKD:bsorjbu) ([raw file](https://github.com/harness/harness-core/blob/8e05c8a68572f712bcc7c5dfb46a7546f9a8afbb/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L272)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Can we add a check here before referencing index 0
</blockquote></details>

Added a validation before this logic
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 288 at r3](https://reviewable.io/reviews/harness/harness-core/44560#-NPSxeZubaCsQcvOcKf7-r3-288:-NPXVLsDDqCrfcynJpbm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8e05c8a68572f712bcc7c5dfb46a7546f9a8afbb/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L288)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

use equals method
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 290 at r3](https://reviewable.io/reviews/harness/harness-core/44560#-NPSxeZubaCsQcvOcKf7-r3-290:-NPXVMXvAj7ev9joneTx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8e05c8a68572f712bcc7c5dfb46a7546f9a8afbb/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L290)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

again a check before referencing index 0
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 522 at r3](https://reviewable.io/reviews/harness/harness-core/44560#-NPSxeZubaCsQcvOcKf7-r3-522:-NPXVOdTE8T8mHDhtwkI:b60rozl) ([raw file](https://github.com/harness/harness-core/blob/8e05c8a68572f712bcc7c5dfb46a7546f9a8afbb/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L522)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

check for size of list here as well
</blockquote></details>

added validation before

**Review by piyush-bhuwalka — APPROVED** [review]
> Reviewed 3 of 13 files at r1, 4 of 9 files at r2, 2 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44560)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**